stardog/files/stardog-wrapper.sh

```diff
@@ -0,0 +1,164 @@
+#!/bin/bash
+
+# saner programming env: these switches turn some bugs into errors
+set -o errexit -o pipefail -o noclobber -o nounset
+
+# -allow a command to fail with !’s side effect on errexit
+# -use return value from ${PIPESTATUS[0]}, because ! hosed $?
+! getopt --test > /dev/null
+if [[ ${PIPESTATUS[0]} -ne 4 ]]; then
+    >&2 echo 'I’m sorry, `getopt --test` failed in this environment.'
+    exit 1
+fi
+
+OPTIONS=hv
+LONGOPTS=disable-user:,set-admin-password,help,ensure-user:,user-pass-from:,new-database:,ensure-role:,ensure-role-grants:,assign-role-to-user:,verbose
+
+# -regarding ! and PIPESTATUS see above
+# -temporarily store output to be able to check for errors
+# -activate quoting/enhanced mode (e.g. by writing out “--options”)
+# -pass arguments only via   -- "$@"   to separate them correctly
+! PARSED=$(getopt --options=$OPTIONS --longoptions=$LONGOPTS --name "$0" -- "$@")
+if [[ ${PIPESTATUS[0]} -ne 0 ]]; then
+    # e.g. return value is 1
+    #  then getopt has complained about wrong arguments to stdout
+    exit 2
+fi
+# read getopt’s output this way to handle the quoting right:
+eval set -- "$PARSED"
+
+function usage {
+	cat <<EOM
+Usage: $(basename "$0") [OPTION]...
+
+     --disable-user        USER                 Disable the given user.
+     --set-admin-password                       Sets the admin password. Needs \$STARDOG_ADMIN_PW set as ENV var.
+     --ensure-user         USER                 Ensures a user exists with the given name. Needs --user-pass-from flag.
+                                                Accepts additional options after -- (e.g. '-- -s' for superuser)
+     --user-pass-from      PATH                 When using --ensure-user, set a path to a file which contains the user password.
+     --ensure-role         ROLE                 Ensures a role exists with the given name.
+     --ensure-role-grants  ROLE,ACTION,OBJECT   Ensures an existing role has the given permissions and grant objects (e.g. role_name,write,db:*)
+     --assign-role-to-user ROLE,USER
+     --new-database        VALUE                Creates a new database with the given name. Pass additional options after --,
+                                                e.g. "--new-database stardog -- -o search.enabled=true"
+                                                Will skip existing databases.
+  -h,--help                                     Display this help and exit.
+  -v,--verbose                                  Verbose mode.
+EOM
+	exit 2
+}
+
+admin_bin="${STARDOG_ADMIN_BIN:-/opt/stardog/bin/stardog-admin}"
```

Comment:
A bash function could be used to always use `--passwd`:
```
stardog-admin () {
  $admin_bin $@ --passwd ${admin_pw}
}
```

Not sure where exactly the argument goes...

Reply:
Hm, this would break scenarios where I need arguments at the end of the command to provide additional args for the command itself, e.g. `bash ${script} -v --new-database my-database -- -o "search.enabled=true"`

Reply:
true, never mind then